packages/expression-parser/src/expression-parser/BooleanExpressionParser.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { ExpressionParser } from './ExpressionParser';
+
+export class BooleanExpressionParser extends ExpressionParser {
+  constructor() {
+    super();
+    this.importFunctions();
+  }
+
+  /**
+   * Import any extra functions that we want to support
+   */
+  importFunctions() {
+    this.importMatchRegexFunction();
+  }
+
+  /**
+   * mathjs does not have any built in regex function.
+   * Import this to support matching regex
+   * Usage:
+   * parser.set('x', '5');
+   * parser.evaluate('matchRegex("x", "5")'); -> returns true
+   * parser.evaluate('matchRegex("x", "1")'); -> returns false
+   */
+  importMatchRegexFunction() {
```

Comment:
Supported comparisons in `checkValueSatisfiesCondition,js`
```
const OPERATOR_TO_VALUE_CHECK = {
  '=': (value, target) => value === target,
  '>=': (value, target) => value >= target,
  '<=': (value, target) => value <= target,
  '>': (value, target) => value > target,
  '<': (value, target) => value < target,
  range: (value, target) => target[0] <= value && value <= target[1],
  rangeExclusive: (value, target) => target[0] < value && value < target[1],
  regex: (value, target) => !!value.match(target),
  in: (value, target) => target.includes(value),
};
```

mathjs supports all the above comparisons excepts regex. Luckily it allows us to import new util